airbyte-webapp/src/views/Connection/CatalogDiffModal/components/DiffAccordion.tsx

```diff
@@ -0,0 +1,112 @@
+import { faAngleDown, faAngleRight } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { Disclosure } from "@headlessui/react";
+import classnames from "classnames";
+import { useMemo } from "react";
+import { useIntl } from "react-intl";
+
+import { ImageBlock } from "components";
+
+import { StreamTransform } from "core/request/AirbyteClient";
+
+import { ModificationIcon } from "../../../../components/icons/ModificationIcon";
+import { getSortedDiff } from "../utils/utils";
+import styles from "./DiffAccordion.module.scss";
+import { DiffFieldTable } from "./DiffFieldTable";
+
+interface DiffAccordionProps {
+  streamTransform: StreamTransform;
+}
+
+export const DiffAccordion: React.FC<DiffAccordionProps> = ({ streamTransform }) => {
+  const { formatMessage } = useIntl();
+
+  const { newItems, removedItems, changedItems } = useMemo(
+    () => getSortedDiff(streamTransform.updateStream),
+    [streamTransform.updateStream]
+  );
+
+  // eslint-disable-next-line css-modules/no-undef-class
+  const nameCellStyle = classnames(styles.nameCell, styles.row);
+  // eslint-disable-next-line css-modules/no-undef-class
+  const namespaceCellStyles = classnames(styles.nameCell, styles.row, styles.namespace);
+
+  return (
+    <div className={styles.accordionContainer}>
+      <Disclosure>
+        {({ open }) => (
+          <>
+            <Disclosure.Button className={styles.accordionButton}>
+              <ModificationIcon />
+              <div
```

Comment:
These divs could be extracted together into their own component so that this Accordion component remains pretty small.